app/src/auto_validator/core/admin.py

```diff
@@ -94,26 +96,30 @@ class ValidatorInstanceAdmin(admin.ModelAdmin):
 
 @admin.register(Server)
 class ServerAdmin(admin.ModelAdmin):
-    list_display = ("name", "ip_address", "validatorinstance", "description", "created_at")
-    search_fields = ("name", "ip_address", "validator_instance__subnet_slot__subnet__name")
+    list_display = ("name", "ip_address", "subnet_slot", "validatorinstance_status", "description", "created_at")
+    search_fields = ("name", "ip_address", "validator_instances__subnet_slot__subnet__name")
 
-    def validatorinstance(self, obj):
-        return obj.validator_instances.hotkey if obj.validator_instances else "N/A"
+    def subnet_slot(self, obj):
+        return obj.validator_instances.subnet_slot if obj.validator_instances else "N/A"
 
+    def validatorinstance_status(self, obj):
+        return obj.validator_instances.status if obj.validator_instances else False
 
-@admin.register(Operator)
-class OperatorAdmin(admin.ModelAdmin):
-    list_display = ("name", "discord_id", "email")
-    search_fields = ("name", "discord_id", "email")
+    validatorinstance_status.boolean = True
 
 
-@admin.register(Block)
-class BlockAdmin(admin.ModelAdmin):
-    list_display = ("serial_number", "timestamp")
-    search_fields = ("serial_number",)
+@admin.register(Operator)
+class OperatorAdmin(admin.ModelAdmin):
+    list_display = ("name", "discord_id")
+    search_fields = ("name", "discord_id")
 
 
 @admin.register(Hotkey)
 class HotkeyAdmin(admin.ModelAdmin):
     list_display = ("hotkey", "is_mother")
     search_fields = ("hotkey",)
+
+    def delete_queryset(self, request: HttpRequest, queryset: QuerySet[Any]) -> None:
+        if queryset.filter(validator_instances__isnull=False).exists():
+            raise Exception("Cannot delete Hotkey because related ValidatorInstance objects exist.")
```

Comment:
this sounds like it's reimplemented `on_delete=models.PROTECT` in the foreignkey of the model